R/calcPastureSuit.R

```diff
@@ -29,88 +29,80 @@ calcPastureSuit <- function(climatetype = "MRI-ESM2-0:ssp126",
+  popDensity <- (population * 1e6) / landArea * 10000 # population density in number of people per km2
   popDensity[is.infinite(popDensity)] <- 0
   popDensity[is.nan(popDensity)] <- 0
 
   years <- intersect(getYears(popDensity), getYears(cellPrep))
 
   aridity <- cellPrep[, years, ] / cellPet[, years, ]
   aridity[is.infinite(aridity) | is.nan(aridity)] <- 0
+
   # 0.5 aridity threshold for managed pastures. Same from HYDE 3.2.
   aridity[aridity < 0.5] <- 0
   aridity[aridity >= 0.5] <- 1
 
   # pasture suitability check
   pastureSuit <- aridity
   popDensity <- popDensity[, getYears(pastureSuit), ]
+
+  cellorder <- match(dimnames(pastureSuit)$x.y.iso, dimnames(popDensity)$x.y.iso)
+  popDensity <- popDensity[cellorder]
   pastureSuit[popDensity < 5] <- 0 # 5 hab km2 population threshold for managed pastures. Same from HYDE 3.2.
 
   pastureSuitArea <- (pastureSuit * landArea * 100) / 1e6 # (from km2 (x100) to mha (/1e6))
-  pastureSuitArea <- collapseDim(pastureSuitArea)
-  pastureSuitArea <- toolHoldConstantBeyondEnd(pastureSuitArea)
 
-  # calibration to historical values
-  histPastr <- setNames(calcOutput("LanduseInitialisation", cellular = TRUE,
-                                   nclasses = "nine", aggregate = FALSE)[, , c("past")], "pastr")
-  pastAll <- intersect(getYears(histPastr), getYears(pastureSuitArea))
-
-  pastLy <- findset("past")
-  pastLy <- pastLy[length(pastLy)] # past last year
-  future <- setdiff(getYears(pastureSuitArea), pastAll)
-
-  pastureSuitAreaReg <- dimSums(pastureSuitArea, dim = c("x", "y"))
-  histPastrReg       <- dimSums(histPastr, dim = c("x", "y"))
-  calibReg <- histPastrReg[, pastLy, ] / pastureSuitAreaReg[, pastLy, ]
-  calibReg[is.infinite(calibReg)] <- 1
-  calibReg[is.nan(calibReg)]      <- 0
-  pastureSuitArea[, future, ] <- calibReg * pastureSuitArea[, future, ]
-
-  pastureSuitArea[is.infinite(pastureSuitArea) | is.nan(pastureSuitArea) | is.na(pastureSuitArea)] <- 0
-  pastureSuitArea[pastureSuitArea < 0] <- 0
-  pastureSuitArea[, pastAll, ] <- histPastr[, pastAll, ]
-
-  y <- intersect(pastAll, getYears(pastureSuitArea))
-  pastureSuitArea[, y, ] <- histPastr[, intersect(pastAll, getYears(pastureSuitArea)), ]
-  pastureSuitArea <- toolHoldConstant(pastureSuitArea, findset("time"))
-  pastureSuitArea <- collapseNames(pastureSuitArea)
-  pastureSuitArea[, pastAll, ] <- histPastr[, pastAll, ]
-  #maybe change to: pastureSuitArea[, pastAll, ] <- max(histPastr[, pastAll, ], pastureSuitArea[, pastAll, ]) #nolint
-
-  # Reduce number of grid cells to 59199
-  if (cells == "magpiecell") {
-    pastureSuitArea <- toolCoord2Isocell(pastureSuitArea, cells = cells)
-  }
-
-  return(list(x            = pastureSuitArea,
-              weight       = NULL,
-              unit         = "Mha",
-              description  = "Area suitable for pasture management in mha",
-              isocountries = FALSE))
+  pastureSuitArea <- toolHoldConstantBeyondEnd(pastureSuitArea)
```

Comment:
this seems like a good and simple choice.
Cool that  the calibration is not needed anymore.

Reply:
I moved the calibration to the preloop step